### PR TITLE
GCC 4.8: <regex> to <regex.h>

### LINKED ISSUE #216: Avoid <regex> (for now)
As discussed in #210, we will try to also support GCC 4.8.

For this to work, the only thing we will need to change is the use of `<regex>` inside the library, as it is only implemented as a stub in GCC 4.8.

Possible alternatives are:
- own, simple string matcher for file names
- use another light-weight regex library, e.g. [slre](https://github.com/cesanta/slre)

For reference, [this](https://github.com/openPMD/openPMD-api/pull/217#issuecomment-393532139) is what needs to be solved.

### PR DESCRIPTION
**Pull request by ax3l**
Close #216 Avoid the missing `<regex>` library in GCC 4.8 by using the POSIX C library equivalent.

### CODE REVIEW

File: src/Series.cpp [hunk 1]
```diff
@@ -24,7 +24,18 @@
 #include "openPMD/Series.hpp"
 
 #include <iostream>
-#include <regex>
+
+#if defined(__GNUC__)
+#   if (__GNUC__ == 4 && __GNUC_MINOR__ < 9)
```
Comment (ax3l) [inline src/Series.cpp:29]: I have tested this locally by replacing this test with a single `#   if __GNUC__ < 9`

**Comment by ax3l** [comment]
@C0nsultant ping :)

**Review by anokfireball — APPROVED** [review]
Recompiling the regex on every call to the functor seems excessive. This should have no significant impact compared to the IO that follows, though.

**Comment by ax3l** [comment]
yep. as a side note: that is probably what the `<regex>` `pattern` init is doing as well

**Pull request merged**